sdk/src/loom/SQLite.ls

```diff
@@ -0,0 +1,423 @@
+        SQLITE_ROW          = 100,
+
+        /** 
+         * The operation is now DONE. Mostly commonly seen returned by Statement.step().
+         */
+        SQLITE_DONE         = 101
+    }
+
+    /** 
+     * The possible column data types returned by Statement.columnType() function.
+     */
+    public enum DataType
+    {
+        /** 
+         * 32 bit integer value.
+         */
+        SQLITE_INTEGER  = 1,
+
+        /** 
+         * 32 bit floating point value.
+         */
+        SQLITE_FLOAT    = 2,
+
+        /** 
+         * NULL terminated string value.
+         */
+        SQLITE_TEXT     = 3,
+
+        /** 
+         * Arbitraty byte array value.
+         */
+        SQLITE_BLOB     = 4,
+
+        /** 
+         * Single NULL value.
+         */
+        SQLITE_NULL     = 5
+    }
+
+
+
+    /** 
+     * An open session to an SQLite database file.
+     */
+    public native class Connection
+    {
+        /** 
+         * Connection.open() flag to indicate that the databse should be accessed as READONLY.
+         */
+        public static const FLAG_READONLY:int   = 1;
+
+        /** 
+         * Connection.open() flag to indicate that the databse should be accessed as as READ AND WRITE.
+         */
+        public static const FLAG_READWRITE:int  = 2;
+
+        /** 
+         * Connection.open() flag to indicate that the databse should be CREATED as a new one.
+         */
+        public static const FLAG_CREATE:int     = 4;
+ 
+
+        /**
+         * Called when the backgroundImport() completes.
+         */
+        public static native var onImportComplete:ImportComplete;
+
+
+        /**
+         * Background import interface for an SQLite database. This loads the passed bytes into 
+         * the given database in a background thread, and fires onImportComplete when done.
+         *
+         *  NOTE: The database name must either be located at a to be a valid system 
+         *  writeable path that begins with Path.getWritablePath(), or merely a plain 
+         *  filename in which case it will internally be saved to the system writeable 
+         *  path location.
+         *
+         *  The format of the JSON data is:
+         *      { "tableA": [ {"columnA": value, "columnB": value, ...}, 
+                                 {"columnA": value, "columnB": value, ...}], 
+                  "tableB": [ {"columnA": value, "columnB": value, ...}, 
+                                 {"columnA": value, "columnB": value, ...}], 
+                  ...}
+         *  Where "table*", etc. is the name of the table to insert into, 
+         *  "column*" is the column to insert "value" into
+         *
+         *  @param database Name of the databse to import the data into.
+         *  @param data JSON data to import into the database.
+         *  @return Boolean Whether or not the background import process was successfully kicked off.
+         */
+//TODO: Change over to ByteArray or similar eventually
```

Comment:
We should pass a string and have the bg thread do the JSON parse to save on cycles. Imagine a 10mb json!